app/src/main/java/com/monkeyteam/chimpagne/model/database/ChimpagnePoll.kt

```diff
@@ -0,0 +1,34 @@
+package com.monkeyteam.chimpagne.model.database
+
+import androidx.collection.ArrayMap
+import java.util.UUID
+
+typealias ChimpagnePollId = String
+
+typealias ChimpagnePollOptionId = String
+
+data class ChimpagnePoll(
+    val id: ChimpagnePollId = UUID.randomUUID().toString(),
+    val title: String = "",
+    val query: String = "",
+    val options: Map<ChimpagnePollOptionId, ChimpagnePollOption> = emptyMap(),
+    val votes: Map<ChimpagneAccountUID, ChimpagnePollOptionId> = emptyMap(),
+) {
+  private fun getVotesPerOptions(): Map<ChimpagnePollOptionId, List<ChimpagneAccountUID>> {
+    val votesPerOptions: MutableMap<ChimpagnePollOptionId, List<ChimpagneAccountUID>> = ArrayMap()

```

Comment:
You've forgotten Martin Odersky's course :sob:

Reply:
Yes I broke protocol, but it is a sacrifice I willing to make.